checker.cpp

```diff
@@ -1,22 +1,94 @@
 #include <assert.h>
 #include <iostream>
+#include <string>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
+struct SOC_Warnings {
+  
+  int  range[10] = {0,20,21,24,25,75,76,80,81,100};
```

Comment:
Make this a struct with names for each value instead of array. Otherwise it will be hard to say which is which.